src/Context.php

```diff
@@ -172,17 +172,154 @@ public static function throwIfPrivateApp(string $message): void
+    }
+
+    /**
+     * Logs a message at WARNING level
+     *
+     * @param string $message The message to log
+     * @param array $context  Key/value pairs of contextual information supporting the log statement
+     *
+     * @throws UninitializedContextException
+     */
+    public static function logWarning(string $message, array $context = []): void
+    {
+        self::log($message, LogLevel::WARNING, $context);
+    }
+
+    /**
+     * Logs a message at ERROR level
+     *
+     * @param string $message The message to log
+     * @param array $context  Key/value pairs of contextual information supporting the log statement
+     *
+     * @throws UninitializedContextException
+     */
+    public static function logError(string $message, array $context = []): void
+    {
+        self::log($message, LogLevel::ERROR, $context);
+    }
+
+    /**
+     * Logs a message at ALERT level
+     *
+     * @param string $message The message to log
+     * @param array $context  Key/value pairs of contextual information supporting the log statement
+     *
+     * @throws UninitializedContextException
+     */
+    public static function logAlert(string $message, array $context = []): void
+    {
+        self::log($message, LogLevel::ALERT, $context);
+    }
+
+    /**
+     * Logs a message at EMERGENCY level
+     *
+     * @param string $message The message to log
+     * @param array $context  Key/value pairs of contextual information supporting the log statement
+     *
+     * @throws UninitializedContextException
+     */
+    public static function logEmergency(string $message, array $context = []): void
+    {
+        self::log($message, LogLevel::EMERGENCY, $context);
+    }
+
+    /**
+     * Logs a message at CRITICAL level
+     *
+     * @param string $message The message to log
+     * @param array $context  Key/value pairs of contextual information supporting the log statement
+     *
+     * @throws UninitializedContextException
+     */
+    public static function logCritical(string $message, array $context = []): void
+    {
+        self::log($message, LogLevel::CRITICAL, $context);
+    }
+
+    /**
+     * Log a deprecation message
+     *
+     * @param string $deprecatedFrom A version that will trigger the message once it has been reached
+     * @param string $message        The message to log
+     * @param array $context         Key/value pairs of contextual information supporting the log statement
+     *
+     * @throws UninitializedContextException
+     */
+    public static function logDeprecation(string $deprecatedFrom, string $message, array $context = []): void
+    {
+        if (!preg_match('#^\d+.\d+.\d+$#', $deprecatedFrom)) {
+            throw new \Exception(sprintf('Encountered an invalid version: "%s"', $deprecatedFrom));
+        }
+
+        $currentVersion = Utils::getVersion();
+
+        if (version_compare($currentVersion, $deprecatedFrom, '<')) {
```

Comment:
I think this condition may be inverted - we want to use this method to announce upcoming deprecations, so if the current version comes before the deprecation version, we log it. If we're past the deprecation version, we should throw an exception as a reminder to deal with deprecations.